DDG4/src/Geant4InputHandling.cpp

```diff
@@ -103,14 +103,13 @@ static void collectPrimaries(Geant4PrimaryMap*         pm,
 
   if ( dau )   {
     Geant4Vertex* dv = new Geant4Vertex(*particle_origine);
-    int vid = int(interaction->vertices.size());
```

Comment:
This is the size of a map, seems a bit random to then assign vertices to that key, which may or may not be occupied. So I changed this to use "mask" instead below.